File: app/src/main/java/com/example/flowershop/database/dao/AccountDao.java
package com.example.flowershop.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import com.example.flowershop.database.entity.Account;

import java.util.List;

@Dao
public interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Account account);

    @Update
    void update(Account account);

    @Delete
    void delete(Account account);

    @Query("SELECT * FROM Account WHERE username = :username")
    Account getAccountByUsername(String username);

    @Query("SELECT * FROM Account WHERE username = :username AND password = :password")
    Account login(String username, String password);

    @Query("SELECT * FROM Account")
    LiveData<List<Account>> getAllAccounts();

    @Query("SELECT * FROM Account")
    List<Account> getAllAccountsSync();
}

File: app/src/main/java/com/example/flowershop/database/dao/AccountDao.kt
package com.example.flowershop.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.example.flowershop.database.entity.Account
import kotlinx.coroutines.flow.Flow

@Dao
interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(account: Account)

    @Update
    suspend fun update(account: Account)

    @Delete
    suspend fun delete(account: Account)

    @Query("SELECT * FROM Account WHERE username = :username")
    suspend fun getAccountByUsername(username: String): Account?

    @Query("SELECT * FROM Account WHERE username = :username AND password = :password")
    suspend fun login(username: String, password: String): Account?

    @Query("SELECT * FROM Account")
    fun getAllAccounts(): Flow<List<Account>>
}